lib/archethic/networking/ip_lookup.ex

```diff
@@ -16,10 +18,13 @@ defmodule Archethic.Networking.IPLookup do
     provider = get_provider()
 
     ip =
-      case apply(provider, :get_node_ip, []) do
-        {:ok, ip} ->
-          Logger.info("Node IP discovered by #{provider}")
-          ip
+      with {:ok, ip} <- apply(provider, :get_node_ip, []),
+           true <- provider == Static || Networking.valid_ip?(ip) do
```

Comment:
Even when provider is `Static` we shouldn't allow private IP at production (MIX_ENV=prod) since it will be blocked for mining in pending_transaction_validation.

https://github.com/archethic-foundation/archethic-node/blob/76cefaceccc1cf0291622be10e473cd12d148475/lib/archethic/mining/pending_transaction_validation.ex#L141

Reply:
Please use something like this:

https://github.com/archethic-foundation/archethic-node/blob/76cefaceccc1cf0291622be10e473cd12d148475/config/prod.exs#L210-L219

Reply:
Then you should integrate with :
```elixir
defp should_validate_node_ip? do
    :archethic
    |> Application.get_env(__MODULE__, [])
    |> Keyword.get(:validate_node_ip, false)
  end
```

So if the ip must be validated, then forbid private IP

Reply:
If condition A = should_validate_node_ip? condition B = valid_ip?, then you should implement:

A` + AB